app/src/main/java/com/asfoundation/wallet/ui/iab/AppcoinsRewardsBuyInteract.kt

```diff
@@ -1,14 +1,29 @@
 package com.asfoundation.wallet.ui.iab
 
 import com.appcoins.wallet.bdsbilling.WalletService
+import com.asfoundation.wallet.interact.SmsValidationInteract
 import com.asfoundation.wallet.support.SupportInteractor
+import com.asfoundation.wallet.wallet_blocked.WalletBlockedInteract
+import com.asfoundation.wallet.wallet_validation.WalletValidationStatus
 import io.reactivex.Completable
 import io.reactivex.Single
 import java.util.*
 
 class AppcoinsRewardsBuyInteract(private val inAppPurchaseInteractor: InAppPurchaseInteractor,
                                  private val supportInteractor: SupportInteractor,
-                                 private val walletService: WalletService) {
+                                 private val walletService: WalletService,
+                                 private val walletBlockedInteract: WalletBlockedInteract,
+                                 private val smsValidationInteract: SmsValidationInteract) {
+
+  fun isWalletBlocked() = walletBlockedInteract.isWalletBlocked()
+
+  fun isWalletVerified() =
+      walletService.getWalletAddress()
+          .flatMap {
+            smsValidationInteract.isValid(it)
+                .map { status -> status == WalletValidationStatus.SUCCESS }
+          }
```

Comment:
Why not?
walletService.getWalletAddress()
          .flatMap { smsValidationInteract.isValid(it) }
          .map { status -> status == WalletValidationStatus.SUCCESS }
          .onErrorReturn { true }
Also, since we are repeating, shouldn't the smsValidationInteract have the method called isWalletVerified and the others interactors just call smsValidationInteract.isWalletVerified?